<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>768</width>
    <height>498</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>RepRaptor</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="serialGroup">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>189</width>
      <height>94</height>
     </rect>
    </property>
    <property name="title">
     <string>Serial</string>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QPushButton" name="portsBtn">
       <property name="text">
        <string>Ports</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="serialBox"/>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="connectBtn">
       <property name="text">
        <string>Connect</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="baudbox"/>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="fileBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>190</width>
      <height>136</height>
     </rect>
    </property>
    <property name="title">
     <string>File</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_2">
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="filename">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Filename: </string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QLabel" name="filelines">
       <property name="text">
        <string>Lines:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QProgressBar" name="progressBar">
       <property name="value">
        <number>24</number>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="sendBtn">
       <property name="text">
        <string>Send</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="pauseBtn">
       <property name="text">
        <string>Pause</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="consoleGroup">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>10</y>
      <width>280</width>
      <height>371</height>
     </rect>
    </property>
    <property name="title">
     <string>Console</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_3">
     <item row="0" column="0">
      <widget class="QTextBrowser" name="terminal">
       <property name="acceptDrops">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="clearbtn">
       <property name="text">
        <string>Clear</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="sendtext"/>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="sendbtn">
       <property name="text">
        <string>Send</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="controlBox">
    <property name="geometry">
     <rect>
      <x>210</x>
      <y>10</y>
      <width>261</width>
      <height>311</height>
     </rect>
    </property>
    <property name="title">
     <string>Control</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_4">
     <item row="0" column="8">
      <widget class="QPushButton" name="homexbtn">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>H X</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="xminusbtn">
       <property name="text">
        <string>X-</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QPushButton" name="xplusbtn">
       <property name="text">
        <string>X+</string>
       </property>
      </widget>
     </item>
     <item row="1" column="8">
      <widget class="QPushButton" name="homeybtn">
       <property name="text">
        <string>H Y</string>
       </property>
      </widget>
     </item>
     <item row="2" column="8">
      <widget class="QPushButton" name="homezbtn">
       <property name="text">
        <string>H Z</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0" colspan="4">
      <widget class="QDoubleSpinBox" name="estepspin">
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>150.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="2">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>E tmp</string>
       </property>
      </widget>
     </item>
     <item row="6" column="5" colspan="2">
      <widget class="QPushButton" name="etmpset">
       <property name="text">
        <string>SET</string>
       </property>
      </widget>
     </item>
     <item row="6" column="7" colspan="2">
      <widget class="QPushButton" name="etmpoff">
       <property name="text">
        <string>OFF</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="2">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>B tmp</string>
       </property>
      </widget>
     </item>
     <item row="7" column="5" colspan="2">
      <widget class="QPushButton" name="btmpset">
       <property name="text">
        <string>SET</string>
       </property>
      </widget>
     </item>
     <item row="7" column="7" colspan="2">
      <widget class="QPushButton" name="btmpoff">
       <property name="text">
        <string>OFF</string>
       </property>
      </widget>
     </item>
     <item row="8" column="2" colspan="3">
      <widget class="QPushButton" name="fanonbtn">
       <property name="text">
        <string>FAN ON</string>
       </property>
      </widget>
     </item>
     <item row="8" column="6" colspan="3">
      <widget class="QPushButton" name="fanoffbtn">
       <property name="text">
        <string>FAN OFF</string>
       </property>
      </widget>
     </item>
     <item row="9" column="2" colspan="3">
      <widget class="QPushButton" name="atxonbtn">
       <property name="text">
        <string>ATX ON</string>
       </property>
      </widget>
     </item>
     <item row="9" column="6" colspan="3">
      <widget class="QPushButton" name="atxoffbtn">
       <property name="text">
        <string>ATX OFF</string>
       </property>
      </widget>
     </item>
     <item row="0" column="6">
      <widget class="QPushButton" name="eplusbtn">
       <property name="text">
        <string>E+</string>
       </property>
      </widget>
     </item>
     <item row="0" column="4">
      <widget class="QPushButton" name="zplusbtn">
       <property name="text">
        <string>Z+</string>
       </property>
      </widget>
     </item>
     <item row="1" column="6">
      <widget class="QPushButton" name="ezerobtn">
       <property name="text">
        <string>E0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="centerbtn">
       <property name="text">
        <string>C</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="yplusbtn">
       <property name="text">
        <string>Y+</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="yminusbtn">
       <property name="text">
        <string>Y-</string>
       </property>
      </widget>
     </item>
     <item row="2" column="6">
      <widget class="QPushButton" name="eminusbtn">
       <property name="text">
        <string>E-</string>
       </property>
      </widget>
     </item>
     <item row="2" column="4">
      <widget class="QPushButton" name="zminusbtn">
       <property name="text">
        <string>Z-</string>
       </property>
      </widget>
     </item>
     <item row="6" column="2" colspan="3">
      <widget class="QSpinBox" name="etmpspin">
       <property name="maximum">
        <number>500</number>
       </property>
      </widget>
     </item>
     <item row="7" column="2" colspan="3">
      <widget class="QSpinBox" name="btmpspin">
       <property name="maximum">
        <number>500</number>
       </property>
      </widget>
     </item>
     <item row="5" column="4" colspan="5">
      <widget class="QDoubleSpinBox" name="stepspin">
       <property name="minimum">
        <double>1.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="4" column="4" colspan="4">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Step</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="3">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>E Step</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="4">
      <widget class="QPushButton" name="homeallbtn">
       <property name="text">
        <string>Home All</string>
       </property>
      </widget>
     </item>
     <item row="3" column="4" colspan="5">
      <widget class="QPushButton" name="releasebtn">
       <property name="text">
        <string>Release</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QGroupBox" name="statusGroup">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>270</y>
      <width>166</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Status</string>
    </property>
    <layout class="QGridLayout" name="gridLayout_5">
     <item row="0" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Extruder</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLCDNumber" name="extruderlcd">
       <property name="segmentStyle">
        <enum>QLCDNumber::Outline</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Bed</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLCDNumber" name="bedlcd">
       <property name="segmentStyle">
        <enum>QLCDNumber::Outline</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QCheckBox" name="checktemp">
       <property name="text">
        <string>Check temperature</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <property name="tristate">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>768</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionRecent"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionRecent">
   <property name="text">
    <string>Recent</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Settings</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>portsBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>serialupdate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>61</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>connectBtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>serialconnect()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>61</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>247</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>homexbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>xhome()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>434</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearbtn</sender>
   <signal>clicked()</signal>
   <receiver>terminal</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>624</x>
     <y>335</y>
    </hint>
    <hint type="destinationlabel">
     <x>654</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xplusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>xplus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>327</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sendtext</sender>
   <signal>returnPressed()</signal>
   <receiver>sendbtn</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>369</y>
    </hint>
    <hint type="destinationlabel">
     <x>548</x>
     <y>397</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yminusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>yminus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>146</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yplusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>yplus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>88</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xminusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>xminus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>256</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>189</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zplusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>zplus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>359</x>
     <y>82</y>
    </hint>
    <hint type="destinationlabel">
     <x>356</x>
     <y>-10</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zminusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>zminus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ezerobtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ezero()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>220</x>
     <y>117</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>homezbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>zhome()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>438</x>
     <y>144</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>homeybtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>yhome()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>434</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>383</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eplusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>eplus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>397</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>42</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>eminusbtn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>eminus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>142</y>
    </hint>
    <hint type="destinationlabel">
     <x>475</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>open()</slot>
  <slot>serialupdate()</slot>
  <slot>serialconnect()</slot>
  <slot>xhome()</slot>
  <slot>homeall()</slot>
  <slot>xplus()</slot>
  <slot>xminus()</slot>
  <slot>yplus()</slot>
  <slot>yminus()</slot>
  <slot>zplus()</slot>
  <slot>zminus()</slot>
  <slot>zhome()</slot>
  <slot>eplus()</slot>
  <slot>eminus()</slot>
  <slot>ezero()</slot>
  <slot>yhome()</slot>
 </slots>
</ui>
